test/js-native-api/common.c

```diff
@@ -0,0 +1,49 @@
+#include <js_native_api.h>
+#include "common.h"
+
+#include <stdio.h>
+
+void add_returned_status(napi_env env,
+                         const char* key,
+                         napi_value object,
+                         napi_status expected_status,
```

Comment:
You need a `char* expected_message` here because the `expected_status` is not necessarily `napi_invalid_arg`. By omitting the `char* expected_message` you are essentially causing this function to output "Invalid argument" no matter what `expected_status` is, as long as it agrees with `actual_status`.

Basically, if `expected_status` agrees with `actual_status`, output `expected_message`, which could be "Invalid argument", but could be some other string corresponding to `expected_status`, and if they don't agree, do the `snprintf()`.

Reply:
Another way to look at it is that, since you cannot access the association between `napi_invalid_arg` and `"Invalid argument"`, because `env` is `NULL` and thus you have no access to `napi_get_last_error_info()`, you must establish the association by passing the `napi_status expected_status, const char* expected_message` pair in the parameters by hand. It's not ideal, but it's the only thing you can do when `env` is `NULL`.